integrations.go

```diff
@@ -40,125 +37,22 @@ func (mi *modulesIntegration) processor(event *Event, hint *EventHint) *Event {
 }
 
 func extractModules() map[string]string {
-	extractedModules, err := getModules()
-	if err != nil {
-		Logger.Printf("ModuleIntegration wasn't able to extract modules: %v\n", err)
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		Logger.Printf("The Modules integration is not available in binaries built without module support.")
 		return nil
 	}
-	return extractedModules
-}
-
-func getModules() (map[string]string, error) {
-	if fileExists("go.mod") {
-		return getModulesFromMod()
-	}
-
-	if fileExists("vendor") {
-		// Priority given to vendor created by modules
-		if fileExists("vendor/modules.txt") {
-			return getModulesFromVendorTxt()
-		}
-
-		if fileExists("vendor/vendor.json") {
-			return getModulesFromVendorJSON()
-		}
-	}
-
-	return nil, fmt.Errorf("module integration failed")
-}
-
-func getModulesFromMod() (map[string]string, error) {
-	modules := make(map[string]string)
-
-	file, err := os.Open("go.mod")
-	if err != nil {
-		return nil, fmt.Errorf("unable to open mod file")
+	modules := map[string]string{
+		info.Main.Path: info.Main.Version,
 	}
-
-	defer file.Close()
-
-	areModulesPresent := false
-
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		splits := strings.Split(scanner.Text(), " ")
-
-		if splits[0] == "require" {
-			areModulesPresent = true
-
-			// Mod file has only 1 dependency
-			if len(splits) > 2 {
-				modules[strings.TrimSpace(splits[1])] = splits[2]
-				return modules, nil
-			}
-		} else if areModulesPresent && splits[0] != ")" && splits[0] != "" {
-			modules[strings.TrimSpace(splits[0])] = splits[1]
+	for _, dep := range info.Deps {
+		ver := dep.Version
+		if dep.Replace != nil {
+			ver += fmt.Sprintf(" => %s %s", dep.Replace.Path, dep.Version)
```

Comment:
When the replacement is coming from a local path, there is no version (empty string).

Reply:
As I try with my local modules. It's mark as version `v0.0.0`

![Screen Shot 2563-04-15 at 21 02 14](https://user-images.githubusercontent.com/484530/79346317-67001980-7f5c-11ea-98e2-0d0b49381112.png)

![Screen Shot 2563-04-15 at 20 59 46](https://user-images.githubusercontent.com/484530/79346277-5d76b180-7f5c-11ea-9311-0c104e82297a.png)

Reply:
Or we could make same as `go list -m all` presents:

```
github.com/getsentry/sentry-go v0.0.0 => ./pkg/sentry-go  
```

Reply:
This is my mistake. It should be `dep.Replace.Version` not `dep.Version`.

Reply:
@rhcarvalho Fixed